addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,280 @@
+
+    # ----------------
+    # Business methods
+    # ----------------
+
+    # API methods
+
+    def _l10n_tw_edi_prepare_item_list(self, json_data):
+        item_list = []
+        for line in self.invoice_line_ids.filtered(lambda line: line.display_type == 'product' and not line._get_downpayment_lines()):
+            taxable = line.tax_ids.filtered(lambda t: t.amount >= 5.0)
+            item_list.append(
+                {
+                    "ItemSeq": line.sequence,
+                    "ItemName": line.product_id.name[:100],
+                    "ItemCount": int(line.quantity),
+                    "ItemWord": line.product_uom_id.name[:6],
+                    "ItemPrice": int(line.price_total / line.quantity),
+                    "ItemTaxType": "1" if len(taxable.ids) >= 1 else "3",
+                    "ItemAmount": int(line.price_total),
+                }
+            )
+        json_data.update({"Items": item_list})
+
+    def _l10n_tw_edi_generate_invoice_json(self):
+        if not self.env.company.l10n_tw_edi_ecpay_merchant_id:
+            raise UserError(_("Please fill in the ECpay API information in the Setting!"))
+
+        # prepare json data
+        self.l10n_tw_edi_related_number = base64.urlsafe_b64encode(uuid.uuid4().bytes).decode()
+        json_data = {
+            "MerchantID": self.env.company.l10n_tw_edi_ecpay_merchant_id,
+            "RelateNumber": self.l10n_tw_edi_related_number,
+            "CustomerIdentifier": self.partner_id.vat or "",
+            "CustomerName": self.partner_id.name,
+            "CustomerAddr": self.partner_id.contact_address,
+            "CustomerEmail": self.partner_id.email or '',
+            "CustomerPhone": self.partner_id.phone or '',
+            "ClearanceMark": "",
+            "Print": str(int(self.l10n_tw_edi_is_print or self.partner_id.vat)),
+            "Donation": str(int(self.l10n_tw_edi_is_donate)),
+            "TaxType": "1" if self.l10n_tw_edi_ecpay_is_included_tax == "1" else "3",
+            "SalesAmount": int(self.amount_total),
+            "InvoiceRemark": "Odoo",
+            "InvType": "07",
+            "vat": self.l10n_tw_edi_ecpay_is_included_tax,
+            "CarrierType": self.l10n_tw_edi_carrier_type or "",
+            "CarrierNum": self.l10n_tw_edi_carrier_number if self.l10n_tw_edi_carrier_type in ["2", "3"] else "",
+        }
+        self._l10n_tw_edi_prepare_item_list(json_data)
+
+        if self.l10n_tw_edi_is_donate:
+            json_data.update({"LoveCode": self.l10n_tw_edi_love_code})
+        return json_data
+
+    def _l10n_tw_edi_send(self, json_content):
+        request_url, merchant_id, hashkey, hashIV = self.company_id._l10n_tw_edi_prepare_api_param()
+        # Ensure to lock the records that will be sent, to avoid risking sending them twice.
+        self.env["res.company"]._with_locked_records(self)
+        # prepare EcPayAPI class
+        ecpay_api = EcPayAPI(hashkey, hashIV, merchant_id)
```

Comment:
I find this hard to use, you always need to call _l10n_tw_edi_prepare_api_param in order to create the API object in order to call it.
Wouldn't it be easier to pass the company record to the constructor and from there, get the info?
```suggestion
        EcPayAPI(company_id).call_ecpay_api("/Issue", json_content)
```